src/biguint.rs

```diff
@@ -2255,6 +2266,200 @@ pub(crate) fn to_str_radix_reversed(u: &BigUint, radix: u32) -> Vec<u8> {
     res
 }
 
+#[cfg(u64_digit)]
+pub struct IterU32Digits<'a> {
+    data: &'a [u64],
+    next_is_lo: bool,
+    last_hi_is_zero: bool,
+}
+#[cfg(u64_digit)]
+impl<'a> IterU32Digits<'a> {
+    fn new(data: &'a [u64]) -> Self {
+        let last_hi_is_zero = data
+            .last()
+            .map(|&last| {
+                let last_hi = (last >> 32) as u32;
+                last_hi == 0
+            })
+            .unwrap_or(false);
+        IterU32Digits {
+            data,
+            next_is_lo: true,
+            last_hi_is_zero,
+        }
+    }
+}
+#[cfg(u64_digit)]
+impl<'a> Iterator for IterU32Digits<'a> {
+    type Item = u32;
+    fn next(&mut self) -> Option<u32> {
+        match self.data.split_first() {
+            Some((&first, data)) => {
+                let next_is_lo = self.next_is_lo;
+                self.next_is_lo = !next_is_lo;
+                if next_is_lo {
+                    Some(first as u32)
+                } else {
+                    self.data = data;
+                    if data.is_empty() && self.last_hi_is_zero {
+                        self.last_hi_is_zero = false;
+                        None
+                    } else {
+                        Some((first >> 32) as u32)
+                    }
+                }
+            }
+            None => None,
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.len();
+        (len, Some(len))
+    }
+
+    fn last(self) -> Option<u32> {
+        self.data.last().map(|&last| {
+            if self.last_hi_is_zero {
+                last as u32
+            } else {
+                (last >> 32) as u32
+            }
+        })
+    }
+
+    fn count(self) -> usize {
+        self.len()
+    }
+}
+#[cfg(u64_digit)]
+impl<'a> ExactSizeIterator for IterU32Digits<'a> {
+    fn len(&self) -> usize {
+        self.data.len() * 2 - usize::from(self.last_hi_is_zero)
+    }
+}
+
+#[cfg(not(u64_digit))]
+pub struct IterU32Digits<'a> {
+    it: core::slice::Iter<'a, u32>,
+}
+#[cfg(not(u64_digit))]
+impl<'a> IterU32Digits<'a> {
+    fn new(data: &'a [u32]) -> Self {
+        Self { it: data.iter() }
+    }
+}
+#[cfg(not(u64_digit))]
+impl<'a> Iterator for IterU32Digits<'a> {
+    type Item = u32;
+    fn next(&mut self) -> Option<u32> {
+        self.it.next().map(|&t| t)
```

Comment:
I suggest `cloned()` instead of these maps -- which can be `copied()` when we raise MSRV.
```suggestion
        self.it.next().cloned()
```